From dfdd95a719f22ae023b6f39ae4744943e02cfcd8 Mon Sep 17 00:00:00 2001
From: Gerard Salvatella <gerard.salvatella@toradex.com>
Date: Fri, 23 Nov 2018 16:37:20 +0100
Subject: [PATCH] ARM: dts: imx7-colibri: use static MTD partition layout

Use static MTD partition layout so that we are not dependent
on U-Boot providing proper MTD partition information.

Signed-off-by: Gerard Salvatella <gerard.salvatella@toradex.com>
---
 arch/arm/boot/dts/imx7-colibri.dtsi | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/arch/arm/boot/dts/imx7-colibri.dtsi b/arch/arm/boot/dts/imx7-colibri.dtsi
index c9e50f1149d4..36299ad816af 100644
--- a/arch/arm/boot/dts/imx7-colibri.dtsi
+++ b/arch/arm/boot/dts/imx7-colibri.dtsi
@@ -142,11 +142,34 @@
 };
 
 &gpmi {
+	#address-cells = <1>;
+	#size-cells = <1>;
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_gpmi_nand>;
 	fsl,use-minimum-ecc;
 	nand-on-flash-bbt;
 	nand-ecc-mode = "hw";
+
+	partition@0 {
+               label = "mx7-bcb";
+               reg = <0x00000000 0x00080000>;
+	};
+	partition@80000 {
+	       label = "u-boot1";
+	       reg = <0x00080000 0x00180000>;
+	};
+	partition@200000 {
+	       label = "u-boot2";
+	       reg = <0x00200000 0x00180000>;
+	};
+	partition@380000 {
+	       label = "u-boot-env";
+	       reg = <0x00380000 0x00080000>;
+	};
+	partition@400000 {
+	       label = "ubi";
+	       reg = <0x00400000 0x1fc00000>;
+	};
 };
 
 &i2c1 {
-- 
2.13.6

